### PR TITLE
systemd: Simplify unit script names

### PR DESCRIPTION
**Pull request by kirelagin**
Make journalctl logs readable again!

* Remove the superfluous `unit-script` prefix. Personally I can tell it’s a script by its distinctive `-start` / `-preStart` / etc. suffix.
* Put it back into `/bin/` (https://github.com/NixOS/nixpkgs/commit/adba92b5ef37e2fb873c29a07d7760119c8701b7 😒) because I don’t want to see those hashes in my `journalctl` output (this and some services printing their own timestamps makes the actual log message start on 3/4 of my screen, and I am not exaggerating).

cc @lheckemann

### CODE REVIEW

**Comment by kirelagin** [comment]
Oh, I just noticed that @lheckemann added this `unix-script` prefix in the same commit. Are we going to fight? 🤔
I suppose, I can concede either `unit-` or `script-`, but not both. (And to be honest, I still don’t see the need in either of those.)

**Comment by lheckemann** [comment]
see discussion at https://github.com/NixOS/nixpkgs/pull/43534

File: nixos/modules/system/boot/systemd.nix [hunk 1]
```diff
@@ -201,8 +201,13 @@ let
     ];
 
   makeJobScript = name: text:
-    let mkScriptName =  s: "unit-script-" + (replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape s) );
-    in  pkgs.writeTextFile { name = mkScriptName name; executable = true; inherit text; };
+    let
+      scriptName = replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape name);
+      out = pkgs.writeShellScriptBin scriptName ''
```
Comment (zhenyavinogradov) [inline nixos/modules/system/boot/systemd.nix:206]: `writeShellScriptBin` will include the shebang, so you can remove it from `makeJobScript` invocations below

Comment (kirelagin) [reply nixos/modules/system/boot/systemd.nix:206]: Oh, yes, I meant to remove it, thanks.

**Comment by kirelagin** [comment]
Looking at the discussion, it seems that people just like seeing `unit-script` in their `/nix/store` paths, is that right? In this case a solution I see is to use `writeTextFile` and add `unit-script` in the derivation name, but not in the script file name.

(However, personally I would prefer not to do this to keep things simple.)

cc @fpletz @globin @grahamc

**Comment by kirelagin** [comment]
If people insist on keeping the prefix, here is a commit that implements this ^. However, as I said, I would really prefer to drop it.

File: nixos/modules/system/boot/systemd.nix [hunk 2]
```diff
@@ -201,8 +201,23 @@ let
...
+        # to keep the script file name short to avoid cluttering logs.
+        name = "unit-script-${scriptName}";
+        executable = true;
+        destination = "/bin/${scriptName}";
+        text = ''
+          #!${pkgs.runtimeShell} -e
+          ${text}
+        '';
+        checkPhase = ''
+          ${pkgs.stdenv.shell} -n "$out/bin/${scriptName}"
```
Comment (Mic92) [inline nixos/modules/system/boot/systemd.nix:217]: I like this part.

Comment (Mic92) [reply nixos/modules/system/boot/systemd.nix:217]: Seems to be the default of `writeShellscriptBin` anyway.

Comment (kirelagin) [reply nixos/modules/system/boot/systemd.nix:217]: Yes, I copied it from there.

**Comment by kirelagin** [comment]
ping

**Comment by kirelagin** [comment]
👋

**Comment by lheckemann** [comment]
@ofborg test boot

**Comment by lheckemann** [comment]
I'd merge this if the tests pass and nobody voices any objections until then.

**Comment by bjornfor** [comment]
Thanks for fixing this!

**Comment by kirelagin** [comment]
Thanks! If you liked this change, please, help me get NixOS/nixpkgs#87219 merged too :).

**Comment by datafoo** [comment]
Will these changes be back-ported to `nixos-20.03-small` and `nixos-20.03` channels?

**Comment by datafoo** [comment]
Sorry if that sounds stupid. I haven't fully understand how/when changes committed on the master branch of this repo end up in a channel.
I was interested to know if these changes will be visible here: https://github.com/NixOS/nixpkgs-channels/blob/nixos-20.03-small/nixos/modules/system/boot/systemd.nix

**Comment by Mic92** [comment]
We usually backport bugfixes and keep major changes in master. I don't know in which category this change falls. If the maintainer don't want to backport it, it will be available in 20.09.

**Pull request merged**